engine/src/main/java/org/terasology/engine/modes/StateLoading.java

```diff
@@ -243,8 +245,17 @@ public void update(float delta) {
         EngineTime time = (EngineTime) context.get(Time.class);
         long startTime = time.getRealTimeInMs();
         while (current != null && time.getRealTimeInMs() - startTime < 20 && !gameEngine.hasPendingState()) {
-            if (current.step()) {
-                popStep();
+            try {
+                if (current.step()) {
+                    popStep();
+                }
+            } catch (Exception e) {
+                logger.error("Error while loading {}", current, e);
+                String errorMessage = String.format("Failed to load game. There was an error during %s.",
+                        current == null ? "the last part" : current.getMessage());
```

Comment:
If `current.getMessage()` is a key for a translateable string, you probably "just" need to push that through `TranslationSystem::translate`.

In screen layers and widget you'll get it by using `@In` annotation:

```java
    @In
    private TranslationSystem translationSystem;
```

Reply:
Okay, I've made it so the part of StateMainMenu that shows these messages will run them through the translation system.